make_roam/aruba/make_roam_aruba_IAP.py

```diff
@@ -8,17 +8,18 @@
 import time
 import json
 import getpass
+import os
 
 from netmiko import ConnectHandler
 
-# Define parameters
-REPLAY = 2
-MIN_POWER = 2
-MAX_POWER = 21
-STEP_POWER = -3
-COUNTDOWN = 10
-SHOW_ARGS = "EIRP|Channel|BSSID"
-CHANNEL = {1: 149, 2: 154}
+# Dirty hacks to make peace with import from subfolders nightmare
+try:
```

Comment:
Так, никаких костылей такого рода. Пиши импорт относительно корня проекта. Либо относительные импорты с точкой, но первое в небольших проектах предпочтительнее.

```suggestion
from make_roam.aruba import config
```

Напоминаю, что при установке импорта в корень проекта, тебе далее при запуске необходимо будет установить PYTHONPATH в корень проекта для каждого терминала в котором делаешь запуск. Или настроить это в IDE, pycharm например делает это автоматически.

Reply:
Я бы и рад, но есть два ограничения:

1. Я не хочу использовать PYTHONPATH так как проект будет запускаться на самых разных машинах самыми разными людьми. Не хочу заставлять их выполнять какие-то дополнительные манипуляции.
2. Я хочу чтобы можно было запустить как основной модуль, так и отдельно make_roam/aruba/make_roam_aruba_IAP.py
Если использовать относительные импорты с точками в главном модуле, будут проблемы с внутренними импортами в make_roam/aruba/make_roam_aruba_IAP.py, так как текущая рабочая директория будет не make_roam/aruba/

Погуглив тему я не нашёл ничего умнее чем использование конструкции типа
try:
    import config
except ImportError:
    from . import config

Есть варианты как сделать лучше с учётом описанных ограничений?

Reply:
Конечно есть.
1) Заставлять прописывать пользователя дополнительные инструкции абсолютно нормально, обычно это указывается в README.md. Можешь посмотреть популярные опенсорс проекты на питоне и увидишь, что многие авторы указывают кучу доп. инструкций перед запуском. Чтобы запустить тот же фласк, нужно сначала экспортировать две переменные окружения. Тем не менее, если ты хочешь автоматизировать этот процесс у тебя есть несколько опций
- написать скрипт start.sh в котором ты за пользователя пропишешь все переменные, и ему останется лишь запустить этот скрипт перед началом работы
- в каждом файле перед импортами прописать добавление корня проекта в sys.path
`import sys; sys.path.append("путь до корня проекта")`

2) Именно по этому я прошу писать абсолютные импорты, так как они не зависят от того, где ты запускаешь код.